tests/foreman/api/test_activationkey.py

```diff
@@ -50,470 +50,482 @@ def _bad_max_hosts():
+
+    :expectedresults: Activation key is created, and its name is not
+        updated.
+
+    :CaseImportance: Low
+
+    :parametrized: yes
+    """
+    act_key = entities.ActivationKey().create()
+    with pytest.raises(HTTPError):
+        entities.ActivationKey(id=act_key.id, name=new_name).update(['name'])
+    new_key = entities.ActivationKey(id=act_key.id).read()
+    assert new_key.name != new_name
+    assert new_key.name == act_key.name
+
+
+@pytest.mark.tier3
+def test_negative_update_max_hosts():
+    """Create an activation key with ``max_hosts == 1``, then update that
+    field with a string value.
+
+    :id: 3bcff792-105a-4577-b7c2-5b0de4f79c77
+
+    :expectedresults: The update fails with an HTTP 422 return code.
+
+    :CaseImportance: Low
+    """
+    act_key = entities.ActivationKey(max_hosts=1).create()
+    with pytest.raises(HTTPError):
+        entities.ActivationKey(id=act_key.id, max_hosts='foo').update(['max_hosts'])
+    assert act_key.read().max_hosts == 1
+
+
+@pytest.mark.tier2
+def test_positive_get_releases_status_code():
+    """Get an activation key's releases. Check response format.
+
+    :id: e1ea4797-8d92-4bec-ae6b-7a26599825ab
+
+    :expectedresults: HTTP 200 is returned with an ``application/json``
+        content-type
+
+    :CaseLevel: Integration
+    """
+    act_key = entities.ActivationKey().create()
+    path = act_key.path('releases')
+    response = client.get(path, auth=settings.server.get_credentials(), verify=False)
+    status_code = http.client.OK
+    assert status_code == response.status_code
+    assert 'application/json' in response.headers['content-type']
+
+
+@pytest.mark.tier2
+def test_positive_get_releases_content():
+    """Get an activation key's releases. Check response contents.
+
+    :id: 2fec3d71-33e9-40e5-b934-90b03afc26a1
+
+    :expectedresults: A list of results is returned.
+
+    :CaseLevel: Integration
+    """
+    act_key = entities.ActivationKey().create()
+    response = client.get(
+        act_key.path('releases'), auth=settings.server.get_credentials(), verify=False
+    ).json()
+    assert 'results' in response.keys()
+    assert type(response['results']) == list
+
+
+@pytest.mark.tier2
+def test_positive_add_host_collections(module_org):
+    """Associate an activation key with several host collections.
+
+    :id: 1538808c-621e-4cf9-9b9b-840c5dd54644
+
+    :expectedresults:
+
+        1. By default, an activation key is associated with no host
+           collections.
+        2. After associating an activation key with some set of host
+           collections and reading that activation key, the correct host
+           collections are listed.
+
+    :CaseLevel: Integration
+
+    :CaseImportance: Critical
+    """
+    # An activation key has no host collections by default.
+    act_key = entities.ActivationKey(organization=module_org).create()
+    assert len(act_key.host_collection) == 0
+
+    # Give activation key one host collection.
```

Comment:
Just a small suggestion, but could this be tidy up to a for loop once this gets merged?